docs/E2E_ARCHITECTURE.md

```
@@ -0,0 +1,990 @@
+kv.get<T>(scope: string, key: string): Promise<T | null>
+kv.set<T>(scope: string, key: string, value: T): Promise<void>
+kv.list<T>(scope: string, prefix?: string): Promise<T[]>
+kv.delete(scope: string, key: string): Promise<void>
+```
+
+**Storage scopes** (from `src/state/schema.ts`):
+
+| Scope Key | Contents | Access Pattern |
+|-----------|----------|----------------|
+| `mem:sessions` | `Session[]` | list by status, get by id |
+| `mem:obs:{sessionId}` | `RawObservation | CompressedObservation[]` | list all for session |
```

markdownlint-cli2 (0.22.1)

[warning] Table column count
Expected: 3; Actual: 4; Too many cells, extra data will be missing

(MD056, table-column-count)